Titanium.Web.Proxy/ProxyServer.cs

```diff
@@ -520,6 +563,18 @@ public void Start()
                 throw new Exception("Proxy is already running.");
             }
 
+
+            try
+            {
+                if (ProxyEndPoints.Any(x => (x as ExplicitProxyEndPoint).GenericCertificate == null))

```

Comment:
This is strange.
1st: when you always access a property you should use direct cast. This will help other developers to know what the problem is. When you use "as", it may return null and you will get a nullreference exception. If you use direct cast, you will receive an invalid cast exception which is better.
2nd: I think you added the try-catch because of the possible nullreference exceptions (cast problems)
Probaby you wanted to write this:
```
                if (ProxyEndPoints.OfType<ExplicitProxyEndPoint>().Any(x => x.GenericCertificate == null))
                {
                    EnsureRootCertificate();
                }

```